File: idling-resource-dialog-fragment/app/src/androidTest/java/com/sqisland/espresso/idling_resource/dialog_fragment/DialogFragmentIdlingResource.java
package com.sqisland.espresso.idling_resource.dialog_fragment;

import android.support.test.espresso.IdlingResource;
import android.support.v4.app.FragmentManager;

public class DialogFragmentIdlingResource implements IdlingResource {
  private final FragmentManager manager;
  private final String tag;
  private ResourceCallback resourceCallback;

  public DialogFragmentIdlingResource(FragmentManager manager, String tag) {
    this.manager = manager;
    this.tag = tag;
  }

  @Override
  public String getName() {
    return DialogFragmentIdlingResource.class.getName() + ":" + tag;
  }

  @Override
  public boolean isIdleNow() {
    boolean idle = (manager.findFragmentByTag(tag) == null);
    if (idle) {
      resourceCallback.onTransitionToIdle();
    }
    return idle;
  }

  @Override
  public void registerIdleTransitionCallback(ResourceCallback resourceCallback) {
    this.resourceCallback = resourceCallback;
  }
}

File: idling-resource-dialog-fragment/app/src/androidTest/java/com/sqisland/espresso/idling_resource/dialog_fragment/DialogFragmentIdlingResource.kt
package com.sqisland.espresso.idling_resource.dialog_fragment

import androidx.fragment.app.FragmentManager
import androidx.test.espresso.IdlingResource

class DialogFragmentIdlingResource(private val manager: FragmentManager, private val tag: String) : IdlingResource {
  private var resourceCallback: IdlingResource.ResourceCallback? = null

  override fun getName() = DialogFragmentIdlingResource::class.java.name + ":" + tag

  override fun isIdleNow(): Boolean {
    val idle = manager.findFragmentByTag(tag) == null
    if (idle) {
      resourceCallback?.onTransitionToIdle()
    }
    return idle
  }

  override fun registerIdleTransitionCallback(resourceCallback: IdlingResource.ResourceCallback) {
    this.resourceCallback = resourceCallback
  }
}